apisix/utils/batch-processor.lua

```diff
@@ -107,7 +107,7 @@ function execute_func(premature, self, batch)
         end
         return
     end
-
+    self.processed_entries = self.processed_entries + #batch.entries
```

Comment:
please read above code, there have `first_fail` case that meaning sent part of `batch_entries` successfully.

Reply:
and for whole `not ok` case, the failed batch will be discard after exceed `max_retry_count`, in this case, batch will be free too, so it should be counting in `processed_entries` too.